src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -2065,116 +2068,148 @@ bool AdaptDispatch::ResetMode(const DispatchTypes::ModeParams param)
-        enabled = _modes.test(Mode::EraseColor);
+        state = _modes.test(Mode::EraseColor) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::VT200_MOUSE_MODE:
-        enabled = _terminalInput.GetInputMode(TerminalInput::Mode::DefaultMouseTracking);
+        state = _terminalInput.GetInputMode(TerminalInput::Mode::DefaultMouseTracking) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::BUTTON_EVENT_MOUSE_MODE:
-        enabled = _terminalInput.GetInputMode(TerminalInput::Mode::ButtonEventMouseTracking);
+        state = _terminalInput.GetInputMode(TerminalInput::Mode::ButtonEventMouseTracking) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::ANY_EVENT_MOUSE_MODE:
-        enabled = _terminalInput.GetInputMode(TerminalInput::Mode::AnyEventMouseTracking);
+        state = _terminalInput.GetInputMode(TerminalInput::Mode::AnyEventMouseTracking) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::UTF8_EXTENDED_MODE:
-        enabled = _terminalInput.GetInputMode(TerminalInput::Mode::Utf8MouseEncoding);
+        state = _terminalInput.GetInputMode(TerminalInput::Mode::Utf8MouseEncoding) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::SGR_EXTENDED_MODE:
-        enabled = _terminalInput.GetInputMode(TerminalInput::Mode::SgrMouseEncoding);
+        state = _terminalInput.GetInputMode(TerminalInput::Mode::SgrMouseEncoding) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::FOCUS_EVENT_MODE:
-        enabled = _terminalInput.GetInputMode(TerminalInput::Mode::FocusEvent);
+        state = _terminalInput.GetInputMode(TerminalInput::Mode::FocusEvent) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::ALTERNATE_SCROLL:
-        enabled = _terminalInput.GetInputMode(TerminalInput::Mode::AlternateScroll);
+        state = _terminalInput.GetInputMode(TerminalInput::Mode::AlternateScroll) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::ASB_AlternateScreenBuffer:
-        enabled = _usingAltBuffer;
+        state = _usingAltBuffer ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
+        ;
         break;
     case DispatchTypes::ModeParams::XTERM_BracketedPasteMode:
-        enabled = _api.GetSystemMode(ITerminalApi::Mode::BracketedPaste);
+        state = _api.GetSystemMode(ITerminalApi::Mode::BracketedPaste) ? DispatchTypes::DECRPM_Enabled : DispatchTypes::DECRPM_Disabled;
```

Comment:
this is a lot of boilerplate for 99 boolean settings and one that returns something other than the 2 main values

Reply:
I initially used a lambda to transform from bool to the enum and I didn't like it. Less code for sure, but not really any more readable.

Do you have a suggestion how I could do this better?

Reply:
Nah, it's OK